src/build.d

```diff
@@ -564,7 +576,59 @@ string[string] getEnvironment()
     version(OSX) version(X86_64)
         dObjc = true;
 
-    return env;
+    if (env.getDefault("ENABLE_DEBUG", "0") != "0")
```

Comment:
Can we copy the documentation comments for these from the Makefile?  Or maybe put them in the `help` output?  We should probably do the same for the targets as well.  It's not a requirement for this PR, but eventually it'd be nice.

Reply:
Also, some of these are only applicable to posix, right?

Reply:
> Also, some of these are only applicable to posix, right?

Yes, actually I'm not sure all of these modes even work on Posix (i.e. dmd still fails the test suite with `ENABLE_DEBUG` - https://github.com/dlang/dmd/pull/7528).

Reply:
> Can we copy the documentation comments for these from the Makefile? Or maybe put them in the help output?

Done.